pkg/executor/push_test.go

```diff
@@ -307,3 +309,67 @@ func Test_makeTransport(t *testing.T) {
 		})
 	}
 }
+
+var calledExecCommand = false
+var calledCheckPushPermission = false
+
+func setCalledFalse() {
+	calledExecCommand = false
+	calledCheckPushPermission = false
+}
+
+func fakeExecCommand(command string, args ...string) *exec.Cmd {
+	calledExecCommand = true
+	cs := []string{"-test.run=TestHelperProcess", "--", command}
+	cs = append(cs, args...)
+	cmd := exec.Command(os.Args[0], cs...)
+	cmd.Env = []string{"GO_WANT_HELPER_PROCESS=1"}
+	return cmd
+}
+
+func fakeCheckPushPermission(ref name.Reference, kc authn.Keychain, t http.RoundTripper) error {
+	calledCheckPushPermission = true
+	return nil
+}
+
+func TestCheckPushPermissions(t *testing.T) {
+	tests := []struct {
+		Destination           string
+		ShouldCallExecCommand bool
+		ExistingConfig        bool
+	}{
+		{"gcr.io/test-image", true, false},
+		{"gcr.io/test-image", false, true},
+		{"localhost:5000/test-image", false, false},
+		{"localhost:5000/test-image", false, true},
+	}
+
+	execCommand = fakeExecCommand
+	checkRemotePushPermission = fakeCheckPushPermission
+	for _, test := range tests {
+		testName := fmt.Sprintf("%s_ExistingDockerConf_%v", test.Destination, test.ExistingConfig)
+		t.Run(testName, func(t *testing.T) {
+			fs = afero.NewMemMapFs()
+			opts := config.KanikoOptions{
+				Destinations: []string{test.Destination},
+			}
+			if test.ExistingConfig {
+				afero.WriteFile(fs, DockerConfLocation, []byte(""), os.FileMode(0644))
```

Comment:
should this file be deleted after the test execution?

Reply:
Ah good catch, I don't think it's necessary since each iteration of the for loop creates a new temporary mem based FS in lin 352. So the original file in memory is already gone by the time the function executed hence defer would be useless. I've updated the PR and added a defer anyway since it doesn't hurt.